api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StreamingState.java

```diff
@@ -217,17 +227,21 @@ private void pollDataFromKafka() {
         events.forEach(this::rememberEvent);
         if (!events.isEmpty()) {
             addTask(this::streamToOutput);
+            addTask(this::pollDataFromKafka);
+        } else {
+            scheduleTask(this::pollDataFromKafka, POLL_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
```

Comment:
I am not sure I understand the change

Reply:
I will roll it back, it was attempt to optimize something different, not related to the ticket.